tests/integration/TestAdminClient.cc

```diff
@@ -103,3 +103,32 @@ TEST(AdminClient, DuplicatedCreateDeleteTopics)
     }
 }
 
+TEST(AdminClient, deleteRecordsFromTopicPartitions)
+{
+    const Topic     topic     = Utility::getRandomString();
+    const Partition partition = 0;
+
+    std::cout << "[" << Utility::getCurrentTime() << "] Topic[" << topic << "] would be used" << std::endl;
+
+    KafkaTestUtility::CreateKafkaTopic(topic, 5, 3);
+
+    const std::vector<std::tuple<Headers, std::string, std::string>> messages = {
+            {Headers{}, "key1", "value1"},
+            {Headers{}, "key2", "value2"},
+            {Headers{}, "key3", "value3"},
+    };
+
+    // Send the messages
+    KafkaTestUtility::ProduceMessages(topic, partition, messages);
+
+    AdminClient adminClient(KafkaTestUtility::GetKafkaClientCommonConfig());
+    std::cout << "[" << Utility::getCurrentTime() << "] " << adminClient.name() << " started" << std::endl;
+
+    TopicPartitionOffsets tpos;
+    tpos[{topic, partition}] = messages.size();
```

Comment:
A better way might be saving the `offset` when sending the messages (from the metadata), and try with more than 1 partitions. -- not a big problem, I could improve the test after merging the PR.

Reply:
> when sending the messages (from the metadata)
I am not sure what this part means so I leave it as it is currently.

Reply:
no problem.